--- a/kdeprint/cups/cupsdconf2/cups-util.c	2017-12-28 21:20:45.000000000 +0000
+++ b/kdeprint/cups/cupsdconf2/cups-util.c	2017-12-28 21:26:14.000000000 +0000
@@ -167,10 +167,10 @@
 	*/
 
 	snprintf(plain, sizeof(plain), "%s:%s", cupsUser(), pwdstring);
-#if CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2
+#if CUPS_VERSION_MAJOR - 0 <= 1 && CUPS_VERSION_MINOR - 0 < 2
-       httpEncode64_2(encode, sizeof(encode), plain, sizeof(plain));
-#else
        httpEncode64(encode, plain);
+#else
+       httpEncode64_2(encode, sizeof(encode), plain, sizeof(plain));
 #endif
 	snprintf(authstring, sizeof(authstring), "Basic %s", encode);
       }
@@ -430,10 +430,10 @@
 	*/
 
 	snprintf(plain, sizeof(plain), "%s:%s", cupsUser(), pwdstring);
-#if CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2
+#if CUPS_VERSION_MAJOR - 0 <= 1 && CUPS_VERSION_MINOR - 0 < 2
-       httpEncode64_2(encode, sizeof(encode), plain, sizeof(plain));
-#else
        httpEncode64(encode, plain);
+#else
+       httpEncode64_2(encode, sizeof(encode), plain, sizeof(plain));
 #endif
 	snprintf(authstring, sizeof(authstring), "Basic %s", encode);
       }
@@ -512,11 +512,11 @@
   * See if we are accessing localhost...
 	the struct has changed in newer versions - PiggZ (adam@piggz.co.uk)
   */
-#if CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2
+#if CUPS_VERSION_MAJOR - 0 <= 1 && CUPS_VERSION_MINOR - 0 < 2
-        if (!httpAddrLocalhost(http))
-#else
 	if (ntohl(*(int*)&http->hostaddr.sin_addr) != 0x7f000001 &&
       strcasecmp(http->hostname, "localhost") != 0)
+#else
+        if (!httpAddrLocalhost(http))
 #endif
    return (0);
 
portage@d4p ~/trinity-base/kdelibs-3.5.13.1/work/kdelibs-3.5.13.1 $ diff -u kdeprint/cups/ipprequest.cpp*
--- a/kdeprint/cups/ipprequest.cpp	2017-12-28 21:18:41.000000000 +0000
+++ b/kdeprint/cups/ipprequest.cpp	2017-12-28 21:29:35.000000000 +0000
@@ -751,10 +751,7 @@
 	cupsFreeOptions(n, options);
 
 	// find an remove that annoying "document-format" attribute
-#if CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR >= 2
+#if CUPS_VERSION_MAJOR - 0 <= 1 && CUPS_VERSION_MINOR - 0 < 2
-    ipp_attribute_t *attr = ippFindAttribute(request_, "document-format", IPP_TAG_NAME);
-    ippDeleteAttribute(request_, attr);
-#else
 	// (can't use IppDeleteAttribute as older cups doesn't have that)
 	ipp_attribute_t	*attr = request_->attrs;
 	while (attr)
@@ -768,6 +765,9 @@
 		}
 		attr = attr->next;
 	}
+#else
+    ipp_attribute_t *attr = ippFindAttribute(request_, "document-format", IPP_TAG_NAME);
+    ippDeleteAttribute(request_, attr);
 #endif
 }

